src/api/aux.cpp

```diff
@@ -19,72 +19,75 @@ void parse_json_file(std::vector<Token> &tokens, std::vector<Order> &orders, std
-        bool t6 = i.value()["allow_partial_fill"];
-        std::string tt7 = i.value()["sell_amount"];
-        boost::multiprecision::mpz_int t7 = static_cast<boost::multiprecision::mpz_int>(tt7);
-        std::string tt8 = i.value()["buy_amount"];
-        boost::multiprecision::mpz_int t8 = static_cast<boost::multiprecision::mpz_int>(tt8);
-        int t9 = _idx_tokens.at(i.value()["fee"]["token"]);
-        std::string tt10 = i.value()["fee"]["amount"];
-        boost::multiprecision::mpz_int t10 = static_cast<boost::multiprecision::mpz_int>(tt10);
-        int t11 = _idx_tokens.at(i.value()["cost"]["token"]);
-        std::string tt12 = i.value()["cost"]["amount"];
-        boost::multiprecision::mpz_int t12 = static_cast<boost::multiprecision::mpz_int>(tt12);
-
-        orders.push_back(Order(t1,t2,t3,t4,t5,t6,t7,t8,t9,t10,t11,t12));
+    for (auto &o: json_file["orders"].items()) {
+        int key = std::stoi(o.key());
+        int sell_token = _idx_tokens.at(o.value()["sell_token"]);
+        int buy_token = _idx_tokens.at(o.value()["buy_token"]);
+        bool is_sell_order = o.value()["is_sell_order"];
+        bool is_liquidity_order = o.value()["is_liquidity_order"];
+        bool allow_partial_fill = o.value()["allow_partial_fill"];
+        std::string sell_amount_string = o.value()["sell_amount"];
+        boost::multiprecision::mpz_int sell_amount = static_cast<boost::multiprecision::mpz_int>(sell_amount_string);
+        std::string buy_amount_string = o.value()["buy_amount"];
+        boost::multiprecision::mpz_int buy_amount = static_cast<boost::multiprecision::mpz_int>(buy_amount_string);
+        int fee_token = _idx_tokens.at(o.value()["fee"]["token"]);
+        std::string fee_amount_string = o.value()["fee"]["amount"];
+        boost::multiprecision::mpz_int fee_amount = static_cast<boost::multiprecision::mpz_int>(fee_amount_string);
+        int cost_token = _idx_tokens.at(o.value()["cost"]["token"]);
+        std::string cost_amount_string = o.value()["cost"]["amount"];
+        boost::multiprecision::mpz_int cost_amount = static_cast<boost::multiprecision::mpz_int>(cost_amount_string);
+
+        orders.push_back(Order(key, sell_token, buy_token, is_sell_order, is_liquidity_order, allow_partial_fill, sell_amount, buy_amount, fee_token, fee_amount, cost_token, cost_amount));
     }
 
     // For-loop that reads all AMM information
-    for (auto &i: json_file["amms"].items()) {
-        int t1 = std::stoi(i.key());
+    for (auto &am: json_file["amms"].items()) {
+        int key = std::stoi(am.key());
 
         // We will only use ConstantProduct AMMs, so this check ensures that we ignore the rest
-        if (i.value()["kind"] != "ConstantProduct")
+        if (am.value()["kind"] != "ConstantProduct")
             continue;
         
-        int t2[2];
-        boost::multiprecision::mpz_int t3[2];
+        int pools_idx[2];
+        boost::multiprecision::mpz_int pools_reserves[2];
         int counter = 0;
```

Comment:
Since "counter" can't be negative, better options are `unsigned int counter;` or even better `std::size_t counter` (which is usually used for counters).